tests/unit-tests/test_trainer.py

```diff
@@ -115,6 +128,247 @@ def test_trainer_setup_evaluator():
+                                     label_field='label',
+                                     node_feats='feat',
+                                     train_task=False)
+    task_config = GSConfig.__new__(GSConfig)
+    setattr(task_config, "task_weight", 0.75)
+    task_info = TaskInfo(task_type=BUILTIN_TASK_NODE_CLASSIFICATION,
+                         task_id=task_id,
+                         task_config=task_config,
+                         dataloader=dataloader)
+    node_feats = np_data.get_node_feats(target_idx, 'feat')
+    labels = np_data.get_node_feats(target_idx, 'label')
+    mini_batch = (target_idx, target_idx, None)
+
+    blocks, input_feats, _, lbl, input_nodes = \
+        prepare_node_mini_batch(np_data, task_info, mini_batch, device)
+    assert blocks is None
+    assert_equal(input_nodes["n1"].numpy(), target_idx["n1"].numpy())
+    assert_equal(input_feats["n1"].cpu().numpy(), node_feats["n1"].numpy())
+    assert_equal(lbl["n1"].cpu().numpy(), labels["n1"].numpy())
+
+    dataloader = GSgnnNodeDataLoader(np_data, target_idx, [10], 10,
+                                     label_field='label',
+                                     train_task=False)
+    task_info = TaskInfo(task_type=BUILTIN_TASK_NODE_CLASSIFICATION,
+                         task_id=task_id,
+                         task_config=task_config,
+                         dataloader=dataloader)
+    _, input_feats, _, lbl, input_nodes = \
+        prepare_node_mini_batch(np_data, task_info, mini_batch, device)
+    assert_equal(input_nodes["n1"].numpy(), target_idx["n1"].numpy())
+    assert len(input_feats) == 0
+    assert_equal(lbl["n1"].cpu().numpy(), labels["n1"].numpy())
+
+def test_mtask_prepare_edge_mini_batch():
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        # get the test dummy distributed graph
+        g, part_config = generate_dummy_dist_graph(graph_name='dummy', dirname=tmpdirname)
+        ep_data = GSgnnData(part_config=part_config)
+
+    setup_device(0)
+    device = get_device()
+
+    target_idx = {('n0', 'r1', 'n1'): th.arange(ep_data.g.number_of_edges('r1'))}
+    task_id = "test_edge_prediction"
+    dataloader = GSgnnEdgeDataLoader(ep_data, target_idx, [10], 10,
+                                     node_feats='feat',
+                                     label_field='label',
+                                     train_task=True, remove_target_edge_type=False)
+
+    task_config = GSConfig.__new__(GSConfig)
+    setattr(task_config, "task_weight", 0.71)
+    task_info = TaskInfo(task_type=BUILTIN_TASK_EDGE_REGRESSION,
+                         task_id=task_id,
+                         task_config=task_config,
+                         dataloader=dataloader)
+    input_node_idx = {
+        "n0": th.arange(10),
+        "n1": th.arange(20),
+    }
+    node_feats = ep_data.get_node_feats(input_node_idx, 'feat')
+    labels = ep_data.get_edge_feats(target_idx, 'label')
+    print(g.edges[('n0', 'r1', 'n1')])
```

Comment:
Is this print line for debug. May remove it.

Reply:
removed